bin/node/src/commands/start.rs

```diff
@@ -0,0 +1,75 @@
+use std::time::Duration;
+
+use anyhow::{anyhow, Context, Result};
+use miden_node_block_producer::{config::BlockProducerConfig, server as block_producer_server};
+use miden_node_faucet::{config::FaucetConfig, server as faucet_server, utils::build_faucet_state};
+use miden_node_rpc::{config::RpcConfig, server as rpc_server};
+use miden_node_store::{config::StoreConfig, db::Db, server as store_server};
+use tokio::task::JoinSet;
+
+use crate::config::NodeConfig;
+
+// START
+// ===================================================================================================
+
+pub async fn start_node(config: NodeConfig) -> Result<()> {
+    let mut join_set = JoinSet::new();
+    let db = Db::setup(config.store.clone().context("Missing store configuration.")?).await?;
```

Comment:
I would rather spawn `start_*` functions here. This would help us to reduce/simplify code.